src/interfaces.ts

```diff
@@ -162,3 +168,26 @@ export interface CommentFlag {
     affectedRange: Range;
     codes: number[] | null;
 }
+
+type ValidateHandler = (scope: Scope, files: (BrsFile | XmlFile)[], callables: CallableContainerMap) => void;
+
+export interface CompilerPlugin {
+    name: string;
+    beforeProgramCreate?: (builder: ProgramBuilder) => void;
+    beforePrepublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
+    afterPrepublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
+    beforePublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
+    afterPublish?: (builder: ProgramBuilder, files: FileObj[]) => void;
+    afterProgramCreate?: (program: Program) => void;
+    beforeProgramValidate?: (program: Program) => void;
+    afterProgramValidate?: (program: Program) => void;
+    beforeTranspile?: (entries: TranspileObj[]) => void;
```

Comment:
can we rename this to `beforeProgramTranspile` and `afterProgramTranspile`?

Reply:
done